.github/workflows/ci-cpp.yaml

```diff
@@ -39,18 +39,18 @@ jobs:
     steps:
       - name: checkout
         uses: actions/checkout@v2
-        with:
-          fetch-depth: 100
-          ref: ${{ github.event.pull_request.head.sha }}
 
-      - name: Check if this pull request only changes documentation
-        id:   docs
-        uses: apache/pulsar-test-infra/diff-only@master
+      - name: Changed files check
+        id:   changes
+        uses: dorny/paths-filter@ca8fa4002c226cbab8b50cc157d3f47899a49762
```

Comment:
I remembered we only can use official Github Actions made by GitHub or a local Github Action. This is a third-party Github Actions and can we use it directly?